python/tests/functional/test_logger.py

```diff
@@ -80,6 +80,19 @@ def test_setup_service_env_var(monkeypatch, root_logger, stdout):
     assert service_name == log["service"]
 
 
+def test_setup_sampling_rate(monkeypatch, root_logger, stdout):
```

Comment:
Could we improve by setting sampling rate to 1 and check if debug logs are always present?

Reply:
I've changed the rate to 1, also added an assertion on log level from stdout to confirm DEBUG level being captured. It might also be interesting to do the opposite as well, with 0 sample rate.

Reply:
okay great, can you also check for the message and we can merge this?

```python
`assert "I am being sampled" == log["message"]`
```

Reply:
> I've changed the rate to 1, also added an assertion on log level from stdout to confirm DEBUG level being captured. It might also be interesting to do the opposite as well, with 0 sample rate.

I thought about it too, but because you're defaulting to 0 we know this works because other tests that capture the INFO value is still working :)